### ISSUE TITLE
Chained select clauses, last one disregarded

### ISSUE THREAD

**Issue by arthur-liberman** [dotnet/efcore#19646]
We have reproduced this issue in EF Core 2.2 and 3.1, on both MSSQL and Oracle (Devart provider)
I am not sure whether we are doing something wrong, or have something configured incorrectly or whether it's an EF Core bug.
We have the following `IQueryable`:

```C#
var query = _context.EntityDbSet
    .OrderBy(nameof(Entity.StartDate))
    .Select(e => new Entity 
       { 
            Id = e.Id, 
            RelatedItemId = e.RelatedItemId, 
            StartDate = e.StartDate, 
            EndDate = e.EndDate ... 
        })
    .Skip(0).Take(10)
```
We then proceed to use this query in several other queries.
The `Id` field is the Key in this table.
The issue I have is as follows:
```C#
var relatedItems = query.Select(e => e.RelatedItemId).ToList();
```

Generates the following SQL:
```sql
SELECT [e].[RELATED_ITEM_ID]
FROM [ENTITY] AS [e]
ORDER BY [e].[START_DATE]
OFFSET @__p_0 ROWS FETCH NEXT @__p_1 ROWS ONLY
```

This works as expected.
But the following doesn't:

```C#
var ids = query.Select(e => e.Id).ToList();
```
Generates the following SQL:
```sql
SELECT [e].[ID], [e].[RELATED_ITEM_ID], [e].[START_DATE], [e].[END_DATE] ... 
FROM [ENTITY] AS [e]
ORDER BY [e].[START_DATE]
OFFSET @__p_0 ROWS FETCH NEXT @__p_1 ROWS ONLY
```

I use `ids.Contains` in a different query. Obviously this fails, because you must only select a single field to use this query in an `IN` SQL clause.

I tried putting the original Select cause at the end of the var query = ... part of the code.
If I completely remove this original select, then it works as expected, selecting only the Id field.

Not sure if this has anything to do with it, but we use a custom value generator for the Id column, 

### Further technical details

EF Core version:
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET Core 2.2, 3.1)
Operating system:
IDE: Visual Studio 2019 16.3

**Comment by ajcvickers** [comment]
@maumar @roji Thoughts on this?

**Comment by maumar** [comment]
@arthur-liberman I'm unable to reproduce this. Here is the code I'm running:

```cs
        [ConditionalFact]
        public virtual void Repro19646()
        {
            using (var ctx = new MyContext19646())
            {
                var query = ctx.EntityDbSet
                    .OrderBy(x => x.StartDate)
                    .Select(e => new Entity19646
                    {
                        Id = e.Id,
                        RelatedItemId = e.RelatedItemId,
                        StartDate = e.StartDate,
                        EndDate = e.EndDate
                    })
                    .Skip(0).Take(10);


                var ids = query.Select(e => e.Id).ToList();
            }
        }

        public class MyContext19646 : DbContext
        {
            public DbSet<Entity19646> EntityDbSet { get; set; }

            protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
            {
                optionsBuilder.UseSqlServer(@"Server=.;Database=Repro19646;Trusted_Connection=True;MultipleActiveResultSets=True");
            }
        }

        public class Entity19646
        {
            public int Id { get; set; }
            public int RelatedItemId { get; set; }
            public DateTime StartDate { get; set; }
            public DateTime EndDate { get; set; }
        }
```

And it generates the following sql:

```sql
SELECT [e].[Id]
FROM [EntityDbSet] AS [e]
ORDER BY [e].[StartDate]
OFFSET @__p_0 ROWS FETCH NEXT @__p_1 ROWS ONLY
```

Please either provide full runnable repro or modify the code above so that it reproduces the issue.

**Comment by arthur-liberman** [comment]
Thank you for trying. It took me a while to figure out what causes the problem, although it should've been obvious.
Here's the code that actually reproduces the problem:
```C#
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using System.Linq;
using System;
using Microsoft.Extensions.Logging;
using System.Collections.Generic;
using System.Threading.Tasks;
using System.Linq.Expressions;

namespace efcoretest
{
    class Program
    {
        static async Task Main(string[] args)
        {
            var a = new test();
            var affected = await a.populate();
            var list = await a.Repro19646();
        }
    }

    public class test
    {
        public async Task<int> populate()
        {
            int result = 0;
            using (var ctx = new MyContext19646())
            {
                if (!ctx.EntityDbSet.Any())
                {
                    for (int i = 0; i < 20; i++)
                    {
                        var e = new Entity19646();
                        e = ctx.EntityDbSet.Add(e).Entity;
                    }
                    result = await ctx.SaveChangesAsync();
                }
            }
            return result;
        }

        public async Task<List<int>> Repro19646()
        {
            using (var ctx = new MyContext19646())
            {
                List<string> fields = new List<string> { "RelatedItemId", "EndDate" };
                if (!fields.Contains(nameof(Entity.Id)))
                    fields.Add(nameof(Entity.Id));
                var query = ctx.EntityDbSet
                    .OrderBy(x => x.StartDate)
                    .Select(CreateNewStatement<Entity19646>(fields))
                    // .Select(o => new Entity19646
                    // {
                    //     Id = o.Id,
                    //     RelatedItemId = o.RelatedItemId,
                    //     EndDate = o.EndDate
                    // })
                    .Skip(0).Take(10);

                var a = query.Select(e => e.Id);
                var b = await ctx.EntityDbSet.Where(e => a.Contains(e.Id)).ToListAsync();
                var c = await query.Where(e => e.Id > 2).ToListAsync();
                return await a.ToListAsync();
            }
        }

        private static Expression<Func<TEntity, TEntity>> CreateNewStatement<TEntity>(IList<string> fields) where TEntity : class
        {
            // input parameter "o"
            var xParameter = Expression.Parameter(typeof(TEntity), "o");

            // new statement "new TEntity()"
            var xNew = Expression.New(typeof(TEntity));

            // create initializers
            var bindings = fields.Select(o => o.Trim())
            .Where(o => typeof(TEntity).GetProperty(o) != null)
            .Select(o =>
            {
                // property "Field1"
                var mi = typeof(TEntity).GetProperty(o);

                // original value "o.Field1"
                var xOriginal = Expression.Property(xParameter, mi);

                // set value "Field1 = o.Field1"
                return Expression.Bind(mi, xOriginal);
            }
            );

            // initialization "new TEntity { Field1 = o.Field1, Field2 = o.Field2 }"
            var xInit = Expression.MemberInit(xNew, bindings);

            // expression "o => new TEntity { Field1 = o.Field1, Field2 = o.Field2 }"
            var lambda = Expression.Lambda<Func<TEntity, TEntity>>(xInit, xParameter);
            return lambda;
        }
    }

    public class MyContext19646 : DbContext
    {
        public static readonly ILoggerFactory loggerFactory = LoggerFactory.Create(builder =>
        {
            builder.AddConsole();
        });

        public DbSet<Entity19646> EntityDbSet { get; set; }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.ApplyConfiguration(new Entity19646Configuration());
        }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder
                .UseLoggerFactory(loggerFactory)
                .EnableSensitiveDataLogging()
                .UseSqlServer(@"Server=myd-hvm00221;Database=Repro19646;User Id=sa;Password=W3lcome1");
        }
    }

    public class Entity19646Configuration : IEntityTypeConfiguration<Entity19646>
    {
        public void Configure(EntityTypeBuilder<Entity19646> builder)
        {
            builder.ToTable("Entity19646");
            builder.HasKey(o => o.Id);
            builder.Property(t => t.Id).ValueGeneratedOnAdd().HasColumnName("id");
            builder.Property(t => t.RelatedItemId).HasColumnName("relatedItemId");
            builder.Property(t => t.StartDate).HasColumnName("startDate");
            builder.Property(t => t.EndDate).HasColumnName("endDate");
        }
    }

    public class Entity
    {
        private int id;

        public int Id { get => id; set => id = value; }
    }

    public class Entity19646 : Entity
    {
        public int RelatedItemId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
    }
}
```
Here's the csproj file:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="3.1.1" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="3.1.1" />
    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="3.1.1" />
  </ItemGroup>

</Project>
```

Note the `Select` method in the sample code. Both the 'active' and commented out methods produce an identical expression when looking at it in during debug, but it seems that something is wrong when we use our implementation, because it fails.
We are using the `CreateNewStatement` method because we have a generic infrastructure, and in our code the domain object properties have private setters (unlike the example above). The reason we create a new instance of the entity class is because we also have DTO mapper classes which are strongly typed and only accept a domain object for a specific DTO.

It's possible that the way `CreateNewStatement` is written is the issue here, but I don't see any obvious problems there.

**Comment by maumar** [comment]
@arthur-liberman looks like this is fixed on the current master. I'm getting the expected sql when using both approaches:

```sql
SELECT [e].[id]
FROM [Entity19646] AS [e]
ORDER BY [e].[startDate]
OFFSET @__p_0 ROWS FETCH NEXT @__p_1 ROWS ONLY
```

**Comment by arthur-liberman** [comment]
@maumar interesting. Is there a way for me to test this? And will there be a fix for this in an upcoming 3.1.x release?

**Comment by maumar** [comment]
@arthur-liberman you can try our daily builds. The process is described here:
https://github.com/dotnet/aspnetcore/blob/master/docs/DailyBuilds.md

**Comment by arthur-liberman** [comment]
@maumar I'm not sure what I'm doing wrong, but it's not working for me.
I'm using VSCode right now for testing this.
I downloaded the 5.0 Master binaries as zip and unzipped it into `F:\dotnetcore\dotnet-sdk-latest-win-x64`.
Should I be using the 3.1.x binaries instead? I'll try it later to be sure.
My csproj:
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp5.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="5.0.0-alpha1" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="5.0.0-alpha1" />
    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="5.0.0-alpha1" />
  </ItemGroup>
</Project>
```

Output I get:
```
F:\dotnetcore\efcoretest>..\dotnet-sdk-latest-win-x64\dotnet run
F:\dotnetcore\efcoretest\efcoretest.csproj : warning NU1603: efcoretest depends on Microsoft.EntityFrameworkCore.InMemory (>= 5.0.0-alpha1) but Microsoft.EntityFrameworkCore.InMemory 5.0.0-alpha1 was not found. An approximate best match of Microsoft.EntityFrameworkCore.InMemory 5.0.0-alpha1.19459.4 was resolved.
F:\dotnetcore\efcoretest\efcoretest.csproj : warning NU1603: efcoretest depends on Microsoft.EntityFrameworkCore.SqlServer (>= 5.0.0-alpha1) but Microsoft.EntityFrameworkCore.SqlServer 5.0.0-alpha1 was not found. An approximate best match of Microsoft.EntityFrameworkCore.SqlServer 5.0.0-alpha1.19459.4 was resolved.
F:\dotnetcore\efcoretest\efcoretest.csproj : warning NU1603: efcoretest depends on Microsoft.Extensions.Logging.Console (>= 5.0.0-alpha1) but Microsoft.Extensions.Logging.Console 5.0.0-alpha1 was not found. An approximate best match of Microsoft.Extensions.Logging.Console 5.0.0-alpha1.19457.7 was resolved.
F:\dotnetcore\efcoretest\efcoretest.csproj : warning NU1603: Microsoft.EntityFrameworkCore 5.0.0-alpha1.19459.4 depends on System.Collections.Immutable (>= 1.8.0-alpha1.19451.2) but System.Collections.Immutable 1.8.0-alpha1.19451.2 was not found. An approximate best match of System.Collections.Immutable 1.8.0-alpha1.19457.3 was resolved.
F:\dotnetcore\efcoretest\efcoretest.csproj : warning NU1603: Microsoft.EntityFrameworkCore 5.0.0-alpha1.19459.4 depends on System.ComponentModel.Annotations (>= 5.0.0-alpha1.19451.2) but System.ComponentModel.Annotations 5.0.0-alpha1.19451.2 was not found. An approximate best match of System.ComponentModel.Annotations 5.0.0-alpha1.19457.3 was resolved.
F:\dotnetcore\efcoretest\efcoretest.csproj : warning NU1603: Microsoft.EntityFrameworkCore 5.0.0-alpha1.19459.4 depends on System.Diagnostics.DiagnosticSource (>= 5.0.0-alpha1.19451.2) but System.Diagnostics.DiagnosticSource 5.0.0-alpha1.19451.2 was not found. An approximate best match of System.Diagnostics.DiagnosticSource 5.0.0-alpha1.19457.3 was resolved.
F:\dotnetcore\efcoretest\efcoretest.csproj : warning NU1603: efcoretest depends on Microsoft.EntityFrameworkCore.InMemory (>= 5.0.0-alpha1) but Microsoft.EntityFrameworkCore.InMemory 5.0.0-alpha1 was not found. An approximate best match of Microsoft.EntityFrameworkCore.InMemory 5.0.0-alpha1.19459.4 was resolved.
F:\dotnetcore\efcoretest\efcoretest.csproj : warning NU1603: efcoretest depends on Microsoft.EntityFrameworkCore.SqlServer (>= 5.0.0-alpha1) but Microsoft.EntityFrameworkCore.SqlServer 5.0.0-alpha1 was not found. An approximate best match of Microsoft.EntityFrameworkCore.SqlServer 5.0.0-alpha1.19459.4 was resolved.
F:\dotnetcore\efcoretest\efcoretest.csproj : warning NU1603: efcoretest depends on Microsoft.Extensions.Logging.Console (>= 5.0.0-alpha1) but Microsoft.Extensions.Logging.Console 5.0.0-alpha1 was not found. An approximate best match of Microsoft.Extensions.Logging.Console 5.0.0-alpha1.19457.7 was resolved.
F:\dotnetcore\efcoretest\efcoretest.csproj : warning NU1603: Microsoft.EntityFrameworkCore 5.0.0-alpha1.19459.4 depends on System.Collections.Immutable (>= 1.8.0-alpha1.19451.2) but System.Collections.Immutable 1.8.0-alpha1.19451.2 was not found. An approximate best match of System.Collections.Immutable 1.8.0-alpha1.19457.3 was resolved.
F:\dotnetcore\efcoretest\efcoretest.csproj : warning NU1603: Microsoft.EntityFrameworkCore 5.0.0-alpha1.19459.4 depends on System.ComponentModel.Annotations (>= 5.0.0-alpha1.19451.2) but System.ComponentModel.Annotations 5.0.0-alpha1.19451.2 was not found. An approximate best match of System.ComponentModel.Annotations 5.0.0-alpha1.19457.3 was resolved.
F:\dotnetcore\efcoretest\efcoretest.csproj : warning NU1603: Microsoft.EntityFrameworkCore 5.0.0-alpha1.19459.4 depends on System.Diagnostics.DiagnosticSource (>= 5.0.0-alpha1.19451.2) but System.Diagnostics.DiagnosticSource 5.0.0-alpha1.19451.2 was not found. An approximate best match of System.Diagnostics.DiagnosticSource 5.0.0-alpha1.19457.3 was resolved.
warn: Microsoft.EntityFrameworkCore.Model.Validation[10400]
      Sensitive data logging is enabled. Log entries and exception messages may include sensitive application data, this mode should only be enabled during development.
info: Microsoft.EntityFrameworkCore.Infrastructure[10403]
      Entity Framework Core 5.0.0-alpha1.19459.4 initialized 'MyContext19646' using provider 'Microsoft.EntityFrameworkCore.SqlServer' with options: SensitiveDataLoggingEnabled
info: Microsoft.EntityFrameworkCore.Database.Command[20100]
      Executing DbCommand [Parameters=[], CommandType='Text', CommandTimeout='30']
      SELECT CASE
          WHEN EXISTS (
              SELECT 1
              FROM [Entity19646] AS [e]) THEN CAST(1 AS bit)
          ELSE CAST(0 AS bit)
      END
info: Microsoft.EntityFrameworkCore.Infrastructure[10403]
      Entity Framework Core 5.0.0-alpha1.19459.4 initialized 'MyContext19646' using provider 'Microsoft.EntityFrameworkCore.SqlServer' with options: SensitiveDataLoggingEnabled
Unhandled exception. System.InvalidOperationException: Sequence contains more than one element
   at System.Linq.ThrowHelper.ThrowMoreThanOneElementException()
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
   at Microsoft.EntityFrameworkCore.Query.SqlExpressionFactory.In(SqlExpression item, SelectExpression subquery, Boolean negated)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryableMethodTranslatingExpressionVisitor.TranslateContains(ShapedQueryExpression source, Expression item)
   at Microsoft.EntityFrameworkCore.Query.QueryableMethodTranslatingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryableMethodTranslatingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
   at System.Linq.Expressions.MethodCallExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at Microsoft.EntityFrameworkCore.Query.QueryableMethodTranslatingExpressionVisitor.TranslateSubquery(Expression expression)
   at Microsoft.EntityFrameworkCore.Query.RelationalSqlTranslatingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
   at System.Linq.Expressions.MethodCallExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at Microsoft.EntityFrameworkCore.Query.RelationalSqlTranslatingExpressionVisitor.Translate(Expression expression)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryableMethodTranslatingExpressionVisitor.TranslateExpression(Expression expression)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryableMethodTranslatingExpressionVisitor.TranslateLambdaExpression(ShapedQueryExpression shapedQueryExpression, LambdaExpression lambdaExpression)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryableMethodTranslatingExpressionVisitor.TranslateWhere(ShapedQueryExpression source, LambdaExpression predicate)
   at Microsoft.EntityFrameworkCore.Query.QueryableMethodTranslatingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryableMethodTranslatingExpressionVisitor.VisitMethodCall(MethodCallExpression methodCallExpression)
   at System.Linq.Expressions.MethodCallExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at Microsoft.EntityFrameworkCore.Query.QueryCompilationContext.CreateQueryExecutor[TResult](Expression query)
   at Microsoft.EntityFrameworkCore.Storage.Database.CompileQuery[TResult](Expression query, Boolean async)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.CompileQueryCore[TResult](IDatabase database, Expression query, IModel model, Boolean async)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.<>c__DisplayClass12_0`1.<ExecuteAsync>b__0()
   at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQueryCore[TFunc](Object cacheKey, Func`1 compiler)
   at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQuery[TResult](Object cacheKey, Func`1 compiler)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.ExecuteAsync[TResult](Expression query, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryProvider.ExecuteAsync[TResult](Expression expression, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryable`1.GetAsyncEnumerator(CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.ConfiguredCancelableAsyncEnumerable`1.GetAsyncEnumerator()
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)
   at efcoretest.test.Repro19646() in F:\dotnetcore\efcoretest\db.cs:line 62
   at efcoretest.Program.Main(String[] args) in F:\dotnetcore\efcoretest\db.cs:line 18
   at efcoretest.Program.<Main>(String[] args)``

**Comment by roji** [comment]
@arthur-liberman unfortunately our daily builds feed isn't in a great state at the moment... You want to use the latest packages with version 5.0.0-alpha.1.* (note the dot after the alpha!). Your nuget.config should look something like the following:

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <clear />
    <add key="aspnet-extensions" value="https://dotnetfeed.blob.core.windows.net/aspnet-extensions/index.json" />
    <add key="efcore" value="https://dotnetfeed.blob.core.windows.net/aspnet-entityframeworkcore/index.json" />
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
  </packageSources>
</configuration>
```

**Comment by arthur-liberman** [comment]
@roji perfect, thanks! Now it works :)
The only 2 questions I have are 
* Do you happen to know what was the cause of this issue, or which commit fixed it? (just out of curiosity)
* Will the fix be included in a future stable release of 3.1.x?

We stumbled on this problem when the UI "asked" only for some of the fields in the table, rather than all of them. As a work around I had to disable this UI-side filter for time being. It would be great if we could have this fixed, and improve performance in the product.

```
F:\dotnetcore\efcoretest>..\dotnet-sdk-50-latest-win-x64\dotnet build
Microsoft (R) Build Engine version 16.6.0-preview-20071-03+4400fa4e1 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 12.96 sec for F:\dotnetcore\efcoretest\efcoretest.csproj.
  You are using a preview version of .NET Core. See: https://aka.ms/dotnet-core-preview
  efcoretest -> F:\dotnetcore\efcoretest\bin\Debug\netcoreapp5.0\efcoretest.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:21.25

F:\dotnetcore\efcoretest>..\dotnet-sdk-50-latest-win-x64\dotnet run
warn: Microsoft.EntityFrameworkCore.Model.Validation[10400]
      Sensitive data logging is enabled. Log entries and exception messages may include sensitive application data, this mode should only be enabled during development.
info: Microsoft.EntityFrameworkCore.Infrastructure[10403]
      Entity Framework Core 5.0.0-alpha.1.20078.1 initialized 'MyContext19646' using provider 'Microsoft.EntityFrameworkCore.SqlServer' with options: SensitiveDataLoggingEnabled
info: Microsoft.EntityFrameworkCore.Database.Command[20101]
      Executed DbCommand (65ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      SELECT CASE
          WHEN EXISTS (
              SELECT 1
              FROM [Entity19646] AS [e]) THEN CAST(1 AS bit)
          ELSE CAST(0 AS bit)
      END
info: Microsoft.EntityFrameworkCore.Infrastructure[10403]
      Entity Framework Core 5.0.0-alpha.1.20078.1 initialized 'MyContext19646' using provider 'Microsoft.EntityFrameworkCore.SqlServer' with options: SensitiveDataLoggingEnabled
info: Microsoft.EntityFrameworkCore.Database.Command[20101]
      Executed DbCommand (63ms) [Parameters=[@__p_0='0', @__p_1='10'], CommandType='Text', CommandTimeout='30']
      SELECT [e].[id], [e].[endDate], [e].[relatedItemId], [e].[startDate]
      FROM [Entity19646] AS [e]
      WHERE [e].[id] IN (
          SELECT [e0].[id]
          FROM [Entity19646] AS [e0]
          ORDER BY [e0].[startDate]
          OFFSET @__p_0 ROWS FETCH NEXT @__p_1 ROWS ONLY
      )
info: Microsoft.EntityFrameworkCore.Database.Command[20101]
      Executed DbCommand (5ms) [Parameters=[@__p_0='0', @__p_1='10'], CommandType='Text', CommandTimeout='30']
      SELECT [t].[relatedItemId] AS [RelatedItemId], [t].[endDate] AS [EndDate], [t].[id] AS [Id]
      FROM (
          SELECT [e].[id], [e].[endDate], [e].[relatedItemId], [e].[startDate]
          FROM [Entity19646] AS [e]
          ORDER BY [e].[startDate]
          OFFSET @__p_0 ROWS FETCH NEXT @__p_1 ROWS ONLY
      ) AS [t]
      WHERE [t].[id] > 2
      ORDER BY [t].[startDate]
info: Microsoft.EntityFrameworkCore.Database.Command[20101]
      Executed DbCommand (3ms) [Parameters=[@__p_0='0', @__p_1='10'], CommandType='Text', CommandTimeout='30']
      SELECT [e].[id]
      FROM [Entity19646] AS [e]
      ORDER BY [e].[startDate]
      OFFSET @__p_0 ROWS FETCH NEXT @__p_1 ROWS ONLY
```

**Comment by maumar** [comment]
@arthur-liberman we discussed it within the team, but unfortunately this doesn't meet our bar for 3.1.x patch - it only seems to affect the dynamic expression generation, which is quite niche/advanced scenario. So the issue will only be officially fixed in the 5.0 timeframe.
